app/services/data_migrations/backfill_withdrawn_or_declined_for_candidate_by_provider.rb

```diff
@@ -0,0 +1,29 @@
+module DataMigrations
+  class BackfillWithdrawnOrDeclinedForCandidateByProvider
+    TIMESTAMP = 20220120161816
+    MANUAL_RUN = false
+
+    def change
+      audits_joins_sql = <<-SQL.squish
+      INNER JOIN audits ON audits.auditable_type = 'ApplicationChoice'
+      AND audits.auditable_id = application_choices.id
+      AND audits.user_type = 'ProviderUser'
+      AND audits.comment IN ('Declined on behalf of the candidate', 'Withdrawn on behalf of the candidate')
+      SQL
+
+      withdrawn_or_declined_by_provider_applications = ApplicationChoice.joins(audits_joins_sql).where('application_choices.status': %w[declined withdrawn]).distinct
+
+      withdrawn_or_declined_by_other_applications = ApplicationChoice.where.not(id: withdrawn_or_declined_by_provider_applications.pluck(:id)).where(status: %w[declined withdrawn]).distinct
+
+      withdrawn_or_declined_by_provider_applications.each do |application|
```

Comment:
for both those sets we are updating a record at a time which could be costly and take time, how many records exist in prod for each, could we look into bulk updating with no callbacks and no touching instead?

Reply:
Chcked prod, turns out there are quite a lot more non-provider ones!
![image](https://user-images.githubusercontent.com/25597009/150504602-64a2a00d-7ea8-47e9-b365-ba575ac12a69.png)

Will take a look at doing bulk instead

Reply:
It's been modified to now use the `update_all` method, which it turns out [is naturally non-touching](https://www.rubydoc.info/gems/activerecord/4.2.0/ActiveRecord%2FRelation:update_all)